File: self/S2_07052025/ConvertPython/Ex2_Python/Ex2_4.py
import cv2
import matplotlib.pyplot as plt
import numpy as np

img = cv2.imread('waterfall.jpg') / 255.0
gamma = 1.9
enhanced_img = np.power(img, gamma)
plt.figure(figsize=(8, 4))
plt.subplot(1, 2, 1), plt.imshow(cv2.cvtColor((img * 255).astype('uint8'), cv2.COLOR_BGR2RGB)), plt.title('Original Image', fontsize=15), plt.axis('off')
plt.subplot(1, 2, 2), plt.imshow(cv2.cvtColor((enhanced_img * 255).astype('uint8'), cv2.COLOR_BGR2RGB)), plt.title('Contrast Image', fontsize=15), plt.axis('off')
plt.show()
plt.savefig('waterfall_contrast.jpg')

File: S2/Ex2_4.m
%Ex2_4 Contrast Enhancement
clear, clc, close all;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%load test image
Img = im2double(imread('waterfall.jpg'));

%contrast enhancement by changing a 'gamma' characteristic
gamma = 0.2; %gamma=1: unchanged; gamma>1: increase; gamma<1: decrease
enhanced_Img = Img .^ gamma;

%plot image
FS = 15; %fontsize of caption
figure, clf;
subplot(1,2,1), imshow(Img, [0 1]), title('Original Image', "fontsize", FS);
subplot(1,2,2), imshow(enhanced_Img, [0 1]), title('Decrease Contrast', "fontsize", FS);

%save image
saveas(gcf, 'Decrease.png');
